src/mqtt_client.rs

```diff
@@ -567,6 +561,7 @@ where
                     // always fit into it.
                     self.properties_cache
                         .replace(Vec::from_slice(binary_props).unwrap());
+                    // TODO: Figure out how to handle cached responses
```

Comment:
I think the only time we realistically will cache a response is when we receive a `Get` command. Specifcally because when we receive this command, we do two things:
1. Publish the `path` requested by the `Get` command
2. Reply to the `Get` command

I imagine we could simplify this into a single response somehow, where it's either:
1. Get command succeeded, topic posted to the proper path
2. Get command failed, response provided to the ResponseTopic

With this structure, all requests get exactly 1 reply, which means we don't need to bother with caching anymore.

Reply:
Agreed. That's better. But do send the correlation data both cases.

Reply:
The other option would be to always send the get response to the response topic (Ok or Err) and extend the list command to also (or invent a new command) to be an actual re-publication run (as on startup).

Reply:
Opted for:
1. The `Get` command only sends one reply (always has CD) to the response topic.
2. Failure to serialize in the `Get` command (i.e. path is invalid) will result in a generic failure being sent